src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { memo } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoInfo } from '../TodoInfo/TodoInfo';
+
+interface TodoListProps {
+  tempTodo: Todo | null;
+  todos: Todo[];
+  onRemoveTodo: (id: number) => void;
+  onUpdateTodo: (todoId: number, updatedValue: string | boolean) => void;
+  loadingTodoIds: number[];
+}
+
+export const TodoList: React.FC<TodoListProps> = memo(({
+  tempTodo,
+  todos,
+  onRemoveTodo,
+  onUpdateTodo,
+  loadingTodoIds,
+}) => {
+  const loadedTodos = tempTodo
+    ? [...todos, tempTodo]
+    : todos;
```

Comment:
Cool solution, because I created an another TodoItem which shows only when tempTodo is present, nice xD